components/apimgt/org.wso2.carbon.apimgt.common.gateway/src/main/java/org/wso2/carbon/apimgt/common/gateway/util/JWTUtil.java

```diff
@@ -97,47 +98,40 @@ public static String generateHeader(Certificate publicCert, String signatureAlgo
          * {"typ":"JWT", "alg":"[2]", "x5t":"[1]", "x5t":"[1]"}
          * */
         try {
+            X509Certificate x509Certificate = (X509Certificate) publicCert;
+
+            //generate the SHA-1 thumbprint of the certificate
+            MessageDigest digestValue = MessageDigest.getInstance("SHA-1");
+            byte[] der = publicCert.getEncoded();
+            digestValue.update(der);
+            byte[] digestInBytes = digestValue.digest();
+            String publicCertThumbprint = hexify(digestInBytes);
+            String base64UrlEncodedThumbPrint;
+            base64UrlEncodedThumbPrint = java.util.Base64.getUrlEncoder()
+                    .encodeToString(publicCertThumbprint.getBytes("UTF-8"));
+
             StringBuilder jwtHeader = new StringBuilder();
             jwtHeader.append("{\"typ\":\"JWT\",");
             jwtHeader.append("\"alg\":\"");
             jwtHeader.append(getJWSCompliantAlgorithmCode(signatureAlgorithm));
             jwtHeader.append("\",");
+            jwtHeader.append("\"x5t\":\"");
```

Comment:
I remember we decided to use a json library instead of doing a string appending to build the json. shall we do that

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12091/commits/5ca931825782d3665e7e34ad7cc3da9ee185ef8f